packages/cells/src/widget.ts

```diff
@@ -683,11 +684,23 @@ export class Cell<T extends ICellModel = ICellModel> extends Widget {
           this.loadEditableState();
         }
         break;
+      case 'tags':
+        this._setTagsAttribute();
+        break;
       default:
         break;
     }
   }
 
+  private _setTagsAttribute() {
+    const tags: string[] = this.model.getMetadata('tags') ?? [];
+    if (tags.length) {
+      this.node.dataset['tags'] = tags.sort().join(',');
+    } else {
+      delete this.node.dataset['tags'];
+    }
```

Comment:
@bollwyvl highlighted in a gitter discussion that this is not scoped. Indeed most of JupyterLab data attributes use `jp-` prefix, eg. `jp-theme-name`/`jpThemeName`, jp-code-runner/jpCodeRunner etc.